### PR TITLE
Rename docblocks in Routes utilities classes

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Extensions in Visual Studio Code do not have access yet to advanced doc blocks like `class-string`, `list`, `T`, causing type error hints when files are opened. I suggest changing them into the conventional formats.

However, for SA, phpstan uderstands them so I just swapped them to the compatible tags (like `@phpstan-return`).

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

File: system/Commands/Utilities/Routes/AutoRouteCollector.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function __construct(string $namespace, string $defaultController, string
     }
 
     /**
-     * @return list<list<string>>
+     * @return array<int, array<int, string>>
```
Comment (kenjis) [inline system/Commands/Utilities/Routes/AutoRouteCollector.php:38]: PHPStan does not fully support `list`. However, there should be no error.
https://github.com/phpstan/phpstan/issues/6358#issuecomment-1012168140

**Review by MGatner — APPROVED** [review]
Elegant solution. How does this affect CS Fixer?

File: system/Commands/Utilities/Routes/AutoRouteCollector.php (hunk 1 above)
Comment (paulbalandan) [reply system/Commands/Utilities/Routes/AutoRouteCollector.php:38]: Hmm. I did not know that for `list`.

**Comment by paulbalandan** [comment]
> Elegant solution. How does this affect CS Fixer?

php-cs-fixer can still process these, with the different `phpdoc_**` rules.

**Comment by sfadschm** [comment]
Just a feeling, but could we maybe provide a list of "fully" compatible IDEs for development? It seems to me that some adjustments for editors in recent time 
cause more trouble than they resolve (see the phpstorm auto-completion issue) or make workarounds necessary, just because some IDE does not (yet) fully support all docblock Features.

This is not meant against the changes here, they look reasonable, just a general idea 😄

File: system/Commands/Utilities/Routes/AutoRouteCollector.php (hunk 1 above)
Comment (kenjis) [inline system/Commands/Utilities/Routes/AutoRouteCollector.php:38]:
```suggestion
     * @return array<int, array<int, string>>
     * @phpstan-return list<list<string>>
```

File: system/Commands/Utilities/Routes/ControllerMethodReader.php [hunk 2]
```diff
@@ -33,9 +33,9 @@ public function __construct(string $namespace)
     }
 
     /**
-     * @param class-string $class
+     * @phpstan-param class-string $class
      *
-     * @return list<array{route: string, handler: string}>
+     * @return array<int, array{route: string, handler: string}>
```
Comment (kenjis) [inline system/Commands/Utilities/Routes/ControllerMethodReader.php:38]:
```suggestion
     * @return array<int, array{route: string, handler: string}>
     * @phpstan-return list<array{route: string, handler: string}>
```

**Review by MGatner — APPROVED** [review]

**Review by kenjis — APPROVED** [review]

**Pull request merged**